src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicProjectExistsUIContext.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+using Guids = Microsoft.VisualStudio.LanguageServices.Guids;
+using ProjectUtils = Microsoft.VisualStudio.IntegrationTest.Utilities.Common.ProjectUtils;
+
+namespace Roslyn.VisualStudio.IntegrationTests.VisualBasic
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class BasicProjectExistsUIContext : AbstractIntegrationTest
+    {
+        public BasicProjectExistsUIContext(VisualStudioInstanceFactory instanceFactory, ITestOutputHelper testOutputHelper)
+            : base(instanceFactory, testOutputHelper)
+        {
+        }
+
+        public override async Task InitializeAsync()
+        {
+            await base.InitializeAsync().ConfigureAwait(true);
+            VisualStudio.SolutionExplorer.CreateSolution(nameof(BasicProjectExistsUIContext));
+        }
+
+        [WpfFact]
+        public void ProjectContextChanges()
+        {
+            Assert.False(VisualStudio.Shell.IsUIContextActive(Guids.VisualBasicProjectExistsInWorkspaceUIContext));
+
+            VisualStudio.SolutionExplorer.AddProject(new ProjectUtils.Project("TestVisualBasicProject"), WellKnownProjectTemplates.ConsoleApplication, LanguageNames.VisualBasic);
+
+            Assert.True(VisualStudio.Shell.IsUIContextActive(Guids.VisualBasicProjectExistsInWorkspaceUIContext));
+
+            VisualStudio.SolutionExplorer.CloseSolution();
+
+            Assert.False(VisualStudio.Shell.IsUIContextActive(Guids.VisualBasicProjectExistsInWorkspaceUIContext));
```

Comment:
consider: 
1. Cover removal of project...
2. Would it be worth adding some code to verify that the context changed events fire correctly when projects are added / removed?

Reply:
How and where the events fire is somewhat out of scope -- the shell is handling all of that for us. I could entirely imagine the shell batching or changing threading rules and we wouldn't care about that.

Reply:
And as far as close vs. remove: there's no difference in code paths from our perspective.